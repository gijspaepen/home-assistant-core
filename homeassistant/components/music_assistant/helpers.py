"""Provide integration helpers that are aware of the mass integration."""

from __future__ import annotations

import asyncio
from dataclasses import dataclass
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from music_assistant.client import MusicAssistantClient


@dataclass
class MusicAssistantEntryData:
    """Hold Mass data for the config entry."""

    mass: MusicAssistantClient
    listen_task: asyncio.Task
