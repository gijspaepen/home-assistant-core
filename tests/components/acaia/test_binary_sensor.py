"""Test binary sensor for acaia integration."""

from unittest.mock import MagicMock

import pytest
from syrupy import SnapshotAssertion

from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

pytestmark = pytest.mark.usefixtures("init_integration")


BINARY_SENSORS = ("connected", "timer_running")


async def test_binary_sensors(
    hass: HomeAssistant,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
    mock_scale: MagicMock,
) -> None:
    """Test the acaia binary sensors."""

    for binary_sensor in BINARY_SENSORS:
        state = hass.states.get(f"binary_sensor.lunar_123456_{binary_sensor}")
        assert state
        assert state == snapshot(name=f"state_binary_sensor_{binary_sensor}")

        entry = entity_registry.async_get(state.entity_id)
        assert entry
        assert entry == snapshot(name=f"entry_binary_sensor_{binary_sensor}")
