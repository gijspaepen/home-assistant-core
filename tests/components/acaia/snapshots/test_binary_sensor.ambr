# serializer version: 1
# name: test_binary_sensors[entry_binary_sensor_connected]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.lunar_123456_connected',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.CONNECTIVITY: 'connectivity'>,
    'original_icon': None,
    'original_name': 'Connected',
    'platform': 'acaia',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'connected',
    'unique_id': 'aa:bb:cc:dd:ee:ff_connected',
    'unit_of_measurement': None,
  })
# ---
# name: test_binary_sensors[entry_binary_sensor_timer_running]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.lunar_123456_timer_running',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.RUNNING: 'running'>,
    'original_icon': None,
    'original_name': 'Timer running',
    'platform': 'acaia',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'timer_running',
    'unique_id': 'aa:bb:cc:dd:ee:ff_timer_running',
    'unit_of_measurement': None,
  })
# ---
# name: test_binary_sensors[state_binary_sensor_connected]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'connectivity',
      'friendly_name': 'LUNAR_123456 Connected',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.lunar_123456_connected',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_binary_sensors[state_binary_sensor_timer_running]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'running',
      'friendly_name': 'LUNAR_123456 Timer running',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.lunar_123456_timer_running',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
